### ISSUE TITLE
Bad documentation on color palettes

### ISSUE THREAD

**Issue by NathanSalapat** [luanti-org/luanti#11927]
## Problem
I'm using param2 = 'color' and param2 = 'colorfacedir' for blocks and stairs/slab combos. Per the api,

`If the palette has less than 256 pixels, then it is
stretched to contain exactly 256 pixels (after arranging the pixels
to one line). The indexing starts from 0.`

I read this as stretching and blurring the pixels, not as stretching and keeping the same pixels and duplicating them.

## Solutions

The api should say whether the pixels are scaled with or without interpolation. 

## Alternatives

If the scaling should have interpolation consider this a bug report I guess. 0.o

## Additional context
Numbers up top are the param2 values. In a 1x8 palette there is no difference between 110 and 96, but when I scale the palette up in GIMP it works as I would expect it too.
![param2](https://user-images.githubusercontent.com/8930456/147985442-97e186b8-90fa-435f-9616-be8c874d0b26.png)

**Comment by sfan5** [comment]
I don't think this detail is very relevant in the sense that modders shouldn't rely on it.
The documentation could be amended to say the following:
<pre>
If the palette has less than 256 pixels, then it is stretched to contain exactly 256 pixels. How it it scaled is an implementation detail, you should provide a palette with enough pixels.
Indexing starts from 0.
</pre>

**Comment by NathanSalapat** [comment]
Hmm, so further reading led me to discover this in the examples section.
`* 2 (width) x 4 (height) palette, index = 31: the top left corner.
  The palette has 8 pixels, so each pixel is stretched to 32 pixels,
  to ensure the total 256 pixels.`
Which I suppose explains that the pixels are stretched without interpolation. Would it make sense to maybe move that out of the examples section and into the palettes description section.
Now that I know it doesn't really bother me either way.

**Comment by Wuzzy2** [comment]
Wait, the stretching is an "implementation detail"? As far I understood it, the current algorithm just repeats the pixels, which is the simplest and most predictable method. And IMHO also the most reasonable.
If it is true that pixel repetition is used here, it's safe to document it as such and don't confuse the reader with "implementation details". (need to check the code to be sure tho).
